doc/api/cluster.markdown

```diff
@@ -2,16 +2,33 @@
 
     Stability: 2 - Stable
 
-A single instance of Node.js runs in a single thread. To take advantage of
-multi-core systems the user will sometimes want to launch a cluster of Node.js
-processes to handle the load.
+Node.js executes JavaScript code within the context of an event loop. Under the hood, non-blocking socket I/O runs on the main event loop thread while blocking DNS and filesystem I/O runs on a secondary thread that notifies the main thread when it's finished. This concurrency model allows a blocking [I/O bound](https://en.wikipedia.org/wiki/I/O_bound) application to easily distribute work across multiple CPU cores. For more details, see the [libuv](http://docs.libuv.org/en/v1.x/design.html) project.
 
-The cluster module allows you to easily create child processes that
-all share server ports.
+The `cluster` API allows a non-blocking I/O bound application to distribute its work across multiple CPU cores. A Node.js cluster is created when a **master process** forks one or more **worker processes**. By design, workers use the same application code as the master. Conveniently, the `cluster.isMaster` property determines if a running process is the master or a worker.
+
+    const cluster = require('cluster');
+    if (cluster.isMaster) {
+      for (var i = 0; i < numCPUs; i++) {
+        cluster.fork();
+      }
+    } else {
+      // Non-blocking I/O work
+    }
+
+The implementation itself is built directly on top of the
```

Comment:
I like how you've included this information here. However, the paragraph could be written with a little more clarity. Maybe something like this?

The `cluster.fork()` function is built on top of [`child_process.fork()`][]. This allows the master to communicate with and manage each of the workers through an IPC channel.